app/models/merchant.rb

```diff
@@ -2,4 +2,17 @@ class Merchant < ApplicationRecord
   validates_presence_of :name
   has_many :items
   has_many :invoices
+
+  def total_revenue
+    invoices.joins(:transactions, :invoice_items)
+    .merge(Transaction.success)
+    .sum("invoice_items.unit_price * invoice_items.quantity")
+  end
+
+  def date_total_revenue(date)
```

Comment:
there's only subtle differences between these two methods. could we perhaps make a method that finds the revenue based on a certain subset of invoices so we can reuse some of this code?